### PR TITLE
hardcoded-ip-test: Redesign test, add more info on fail

### PR DESCRIPTION
**Pull request by kosstennbl**
## Description
Add information about file location, line numbers and lines themselves if the hardcoded_ip_addresses_in_k8s_runtime_configuration fails

## Issues:
Refs: #2093

## How has this been tested:
Covered by existing integration testing

## Types of changes:
 Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: src/tasks/workload/configuration.cr [hunk 1]
```diff
@@ -286,16 +286,31 @@ task "hardcoded_ip_addresses_in_k8s_runtime_configuration" do |t, args|
...
     end
-
-    ip_search = File.read_lines("#{destination_cnf_dir}/helm_chart.yml").take_while{|x| x.match(/NOTES:/) == nil}.reduce([] of String) do |acc, x|
-      (x.match(/([0-9]{1,3}[\.]){3}[0-9]{1,3}/) &&
-       x.match(/([0-9]{1,3}[\.]){3}[0-9]{1,3}/).try &.[0] != "0.0.0.0" &&
-       x.match(/([0-9]{1,3}[\.]){3}[0-9]{1,3}/).try &.[0] != "127.0.0.1") ? acc << x : acc
+    
+    found_violations = [] of NamedTuple(line_number: Int32, line: String)
+    line_number = 1
+    File.open("#{destination_cnf_dir}/helm_chart.yml") do |file|
```
Comment (martin-mat) [inline src/tasks/workload/configuration.cr:292]: Can #{destination_cnf_dir}/helm_chart.yml be stored in a variable so it does need to be copy/pasted?

File: src/tasks/workload/configuration.cr [hunk 2]
```diff
@@ -286,16 +286,31 @@ task "hardcoded_ip_addresses_in_k8s_runtime_configuration" do |t, args|
...
+          matches.each do |match|
+            unless match.to_s == "0.0.0.0" || match.to_s == "127.0.0.1"
+              found_violations << {line_number: line_number, line: line.strip}
+            end
+          end
+        end
+        line_number += 1
+      end
+    end
+    unless found_violations.empty?
```
Comment (martin-mat) [inline src/tasks/workload/configuration.cr:306]: simplify:
```
    if found_violations.empty?
      CNFManager::TestcaseResult.new(CNFManager::ResultStatus::Passed, "No hard-coded IP addresses found in the runtime K8s configuration")
    else
      stdout_failure("Hard-coded IP addresses found in #{destination_cnf_dir}/helm_chart.yml")
      found_violations.each do |violation|
        stdout_failure("  * Line #{violation[:line_number]}: #{violation[:line]}")
      end
      CNFManager::TestcaseResult.new(CNFManager::ResultStatus::Failed, "Hard-coded IP addresses found in the runtime K8s configuration")
    end

**Review by martin-mat — CHANGES_REQUESTED** [review]

File: src/tasks/workload/configuration.cr (hunk 1 above)
Comment (kosstennbl) [reply src/tasks/workload/configuration.cr]: Yep, it repeated 4 times, good idea, done.

File: src/tasks/workload/configuration.cr (hunk 2 above)
Comment (kosstennbl) [reply src/tasks/workload/configuration.cr]: Felt about test results returns as a "separate code block", but agree that it makes sense to join outputs like this, done.

**Review by collivier — APPROVED** [review]

**Review by Smitholi67 — APPROVED** [review]

**Review by rich-l — APPROVED** [review]
lgtm

File: src/tasks/workload/configuration.cr [hunk 3]
```diff
@@ -281,23 +282,36 @@ task "hardcoded_ip_addresses_in_k8s_runtime_configuration" do |t, args|
...
-       x.match(/([0-9]{1,3}[\.]){3}[0-9]{1,3}/).try &.[0] != "0.0.0.0" &&
-       x.match(/([0-9]{1,3}[\.]){3}[0-9]{1,3}/).try &.[0] != "127.0.0.1") ? acc << x : acc
+    
+    found_violations = [] of NamedTuple(line_number: Int32, line: String)
+    line_number = 1
+    File.open("#{helm_chart_yml_path}") do |file|
+      file.each_line do |line|
+        if line.matches?(/NOTES:/)
+          break
+        elsif matches = line.scan(/([0-9]{1,3}[\.]){3}[0-9]{1,3}/)
```
Comment (svteb) [inline src/tasks/workload/configuration.cr:297]: Can a case where there are multiple IPs on one line even happen?

**Review by svteb — APPROVED** [review]

**Pull request merged**